corehq/apps/accounting/payment_handlers.py

```diff
@@ -409,14 +409,15 @@ def _pay_invoice(self, invoice):
                     autopay_card,
                     amount_in_dollars=amount,
                     description='Auto-payment for Invoice %s' % invoice.invoice_number,
+                    idempotency_key=invoice.id
                 )
+                payment_record.transaction_id = transaction_id
+                payment_record.save()
```

Comment:
I moved it to the `try` block, because I want to achieve: when double charge, stripe will return the same transaction id,  and when we change the second payment record's transaction id to the same transaction id, it will raise KeyError, and all database operations within `transaction.atomic` block will be rolled back. 
> This would mean that there would be no record of this payment, but we would have already charged it via Stripe right? 
Correct. I need to think about it and get back to you later.